solr/core/src/java/org/apache/solr/cloud/RecoveryStrategy.java

```diff
@@ -911,18 +911,39 @@ private final void sendPrepRecoveryCmd(String leaderBaseUrl, String leaderCoreNa
     int readTimeout =
         conflictWaitMs
             + Integer.parseInt(System.getProperty("prepRecoveryReadTimeoutExtraWait", "8000"));
-    try (HttpSolrClient client =
+    var recoveryExec =
+        ExecutorUtil.newMDCAwareFixedThreadPool(
+            1, new SolrNamedThreadFactory("sendPrepRecoveryCmd"));
```

Comment:
Maybe use newMDCAwareSingleThreadExecutor ?  Also define the threadFactory as a field so that the thread names it generates will change over time (increments a counter)